src/App/Fossa/Analyze/Filter.hs

```diff
@@ -8,34 +8,53 @@ import App.Fossa.Config.Analyze (IncludeAll (..))
 import Data.Flag (Flag, fromFlag)
 import Data.List.NonEmpty (NonEmpty, fromList)
 import Srclib.Converter qualified as Srclib
-import Srclib.Types (SourceUnit)
+import Srclib.Types (LicenseSourceUnit (licenseSourceUnitLicenseUnits), LicenseUnit (licenseUnitName), SourceUnit)
 
 data CountedResult
   = NoneDiscovered
   | FilteredAll
-  | FoundSome (NonEmpty SourceUnit)
+  | FoundDependenciesOnly (NonEmpty SourceUnit)
```

Comment:
The final three constructors here are isomorphic to `ScanUnits`. Is there a reason to separate them out this way rather than just
```haskell
data CountedResult
= NoneDiscovered
   | FilteredAll
   | CountedScanUnits ScanUnits
```

Reply:
You're right, that's much nicer. I'll make that change too

Thank you!